drivers/gpio/gpio_sx1509b.c

```diff
@@ -348,6 +537,43 @@ static int sx1509b_init(struct device *dev)
 		goto out;
 	}
 
+#ifdef CONFIG_GPIO_SX1509B_INTERRUPT
+	drv_data->gpio_int = device_get_binding(cfg->gpio_int_dev_name);
+	if (!drv_data->gpio_int) {
+		/* interrupt not supported */
+		drv_data->interrupt_sup = 0;
+	} else {
+		drv_data->interrupt_sup = 1;
+
+		drv_data->work.handler = sx1509b_work_handler;
+
+		gpio_pin_configure(drv_data->gpio_int, cfg->gpio_pin,
+				GPIO_INPUT | GPIO_PULL_UP);
+		gpio_pin_interrupt_configure(drv_data->gpio_int, cfg->gpio_pin,
+				GPIO_INT_EDGE_TO_INACTIVE);
+
+		gpio_init_callback(&drv_data->gpio_cb, sx1509_int_cb,
+				BIT(cfg->gpio_pin));
+		gpio_add_callback(drv_data->gpio_int, &drv_data->gpio_cb);
+		gpio_enable_callback(drv_data->gpio_int, cfg->gpio_pin);
+	}
+#else
+	/* interrupt not supported */
+	drv_data->interrupt_sup = 0;
+#endif
+
+	/* Reset state */
```

Comment:
Remove this and instead add this to the existing initialization after the `k_sleep()` below:

    .interrupt = ALL_PINS,

Reply:
Can you please help me understand this? By default we are enabling all interrupts?

Reply:
The `interrupt` field corresponds to `RegInterruptMask` which powers up with all ones.  Interrupts must be enabled by clearing the corresponding bit.

I hadn't bothered initializing the field properly because we didn't have interrupt support.  Now it needs to be initialized.